### ISSUE TITLE
Mobile view Error in render: "TypeError: _vm.day is undefined"

### ISSUE THREAD

**Issue by whitewaterdesign** [krystalcampioni/vue-hotel-datepicker#269]
**Describe the bug**
version 4.3.0 When you reduce the viewport width to < 768px, the datepicker doesn't open on clicking Check in / Check out and throws the following errors:

Error in render: "TypeError: _vm.day is undefined"
TypeError: _this.$refs.datepickerMonth[0] is undefined

Same issue in the sandbox from GitHub.

App.vue as follows:

```vue
<template>
  <div id="app">
    <HotelDatePicker />
  </div>
</template>

<script>
import HotelDatePicker from 'vue-hotel-datepicker'
import 'vue-hotel-datepicker/dist/vueHotelDatepicker.css';

export default {
  name: 'App',
  components: {
    HotelDatePicker
  }
}
</script>

<style>
</style>
```

Stacktrace below.

**To Reproduce**
1. Set up new project in Vue (2)
2. Follow readme steps to add component
3. Reduce screen width to 767px (using responsive design mode if need be)
2. Click on 'Check-in'
3. Scroll down to Console

**Expected behavior**
Calendar should open in mobile view with no issues

**Desktop (please complete the following information):**
 - OSX 11.0.1
 - FF 86, Safar  14.0.1 i, Chrome 88.0.4324.192

**Additional context**
Stacktrace

```javascript
TypeError: _vm.day is undefined
    render vueHotelDatepicker.common.js:3499
    renderList VueJS
    render vueHotelDatepicker.common.js:3499
    VueJS 14
    set vueHotelDatepicker.common.js:6097
    showDatepicker vueHotelDatepicker.common.js:6808
    toggleDatepicker vueHotelDatepicker.common.js:6811
    VueJS 3
```

```javascript
TypeError: _this.$refs.datepickerMonth[0] is undefined
    set vueHotelDatepicker.common.js:6107
    VueJS 9
    set vueHotelDatepicker.common.js:6097
    showDatepicker vueHotelDatepicker.common.js:6808
    toggleDatepicker vueHotelDatepicker.common.js:6811
    VueJS 3
```

**Comment by whitewaterdesign** [comment]
Issue is here:

src/Datepicker/HotelDatePicker.vue

  169            :belongsToThisMonth="day.belongsToThisMonth"
  175            :date="day.date"

**Comment by matiasperrone** [comment]
Ok, interesting... so day gets undefined in mobile.
Yeah the repetition in HTML markup is not my personal favorite but it works for now, not my doing but I'll try to correct this issue probably today or in the next few days.

**Comment by matiasperrone** [comment]
Removing those lines will compromise some features so... I have to fix it properly.
Do you have any more info or this will reproduce the error?

**Comment by whitewaterdesign** [comment]
Ok, no worries, having the same issue in the sandbox environment.  

There's a switch: 
`v-if="isDesktop || alwaysVisible" `- this <Month/> component is fine.  Next v-if block at line 148:
```
<div
          v-if="!isDesktop && isOpen && !alwaysVisible"
```
activates the mobile `<Month /> `component which has a ` :belongsToThisMonth="day.belongsToThisMonth"` and a `:date="day.date"` prop.

**Comment by apostolisCode** [comment]
Hi,

Can someone provide a quick fix until this is fixed properly?

Thanks

**Comment by develth** [comment]
Issue still exists in 4.4.
Not useable in mobile

**Comment by develth** [comment]
Thanks! Any plans to get the new Version to npm?

**Comment by saniales** [comment]
I still have the issue on v4.4.0

**Comment by develth** [comment]
4.4.0 is the unfixed Version. The fix is in this Repos Tag 4.3.1 which is not released to npm

**Comment by saniales** [comment]
> 4.4.0 is the unfixed Version. The fix is in this Repos Tag 4.3.1 which is not released to npm
why there is a version mismatch? You may also want to fix this
